apis/kueue/v1beta1/resourceflavor_types.go

```diff
@@ -78,6 +79,11 @@ type ResourceFlavorSpec struct {
 	// +optional
 	// +listType=atomic
 	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:validation:XValidation:rule="self.all(x, !has(x.key) ? x.operator == 'Exists' : true)", message="operator must be Exists when 'key' is empty, which means 'match all values and all keys'"
+	// +kubebuilder:validation:XValidation:rule="self.all(x, has(x.tolerationSeconds) ? x.effect == 'NoExecute' : true)", message="effect must be 'NoExecute' when 'tolerationSeconds' is set"
+	// +kubebuilder:validation:XValidation:rule="self.all(x, !has(x.operator) || x.operator in ['Equal', 'Exists'])", message="supported toleration values: 'Equal'(default), 'Exists'"
+	// +kubebuilder:validation:XValidation:rule="self.all(x, has(x.operator) && x.operator == 'Exists' ? !has(x.value) : true)", message="a value must be empty when 'operator' is 'Exists'"
+	// +kubebuilder:validation:XValidation:rule="self.all(x, has(x.effect) ? x.effect in ['NoSchedule', 'PreferNoSchedule', 'NoExecute'] : true)", message="supported taint effect values: 'NoSchedule', 'PreferNoSchedule', 'NoExecute'"
```

Comment:
This is validating something similar than line 84, but they written very differently. Pick one and stick with it.
I kind of prefer `!has(x.operator) || x.operator in ['Equal', 'Exists']`